tests/pages/courses/id.test.js

```diff
@@ -0,0 +1,198 @@
+    getCourse.mockResolvedValue(mockCourse) // Mock getCourse function to return mock course data
+    getFieldContent.mockImplementation((obj, field) => obj[field]) // Mock field content retrieval
+    getAllCohorts.mockResolvedValue([mockCohort]) // Mock getAllCohorts function to return a list of cohorts
+    getCurrentCohort.mockReturnValue(mockCohort) // Mock getCurrentCohort function to return the current cohort
+    getLessonsSubmissions.mockResolvedValue([]) // Mock lesson submissions
+    getUserFromFirestore.mockResolvedValue(mockUser) // Mock user data retrieval from Firestore
+    auth.currentUser = { uid: 'user-123' } // Mock the current user
+
+    global.fetch = jest.fn(() =>
+      Promise.resolve({
+        json: () => Promise.resolve({ data: 'fake data' }), // Mock fetch API to return dummy data
+      })
+    )
+  })
+
+  // Helper function to render the Course component with default or provided props
+  const renderCourse = (props = {}) => {
+    return render(
+      <SessionProvider session={null}>
+        {' '}
+        {/* Wrap the component in SessionProvider for authentication context */}
+        <Course course={mockCourse} currentDate={new Date().getTime()} {...props} />{' '}
+        {/* Pass mock course data */}
+      </SessionProvider>
+    )
+  }
+
+  it('shows NotFound component if course is not active', async () => {
+    renderCourse({ course: { ...mockCourse, active: false } }) // Render with an inactive course
+    await waitFor(() => {
+      expect(screen.getByText('Oops!')).toBeInTheDocument() // Expect the "Oops!" message to be shown
+    })
+  })
+
+  it('renders the course title and description', async () => {
+    renderCourse() // Render the course component
+    expect(screen.getByText('Test Course')).toBeInTheDocument() // Expect course title to be displayed
+    expect(screen.getByText('A test course description')).toBeInTheDocument() // Expect course description to be displayed
+  })
+
+  it('allows user to register for the course', async () => {
+    renderCourse() // Render the course component
+    const registerButton = await screen.findByText('subscribeNow') // Find the subscription button
+    expect(registerButton).toBeInTheDocument() // Ensure the subscription button is present
+    fireEvent.click(registerButton) // Simulate a click on the registration button
+    expect(registerUserInCohortInFirestore).toHaveBeenCalledWith('RU5mLpQrZZWlmftNSB2w', 'user-123') // Ensure user registration is triggered
+  })
+
+  it('shows Calendar buttons after subscribing to the course', async () => {
+    renderCourse() // Render the course component
+
+    const registerButton = await screen.findByText('subscribeNow') // Find the subscription button
+    expect(registerButton).toBeInTheDocument() // Ensure the subscription button is present
+    fireEvent.click(registerButton) // Simulate a click on the registration button
+
+    await waitFor(async () => {
+      const registeredUser = {
```

Comment:
this is being reuse three times at the code. You can define the `registeredUser` in another place to be consumed for all tests that need a registered user, these you've developed and future tests.